src/alchemlyb/constants.py

```diff
@@ -0,0 +1,5 @@
+'''Physical and mathematical constants and units.'''
+# Taken from scipy.constants since py2 doesn't support it
+k = 1.380649e-23
+N_A = 6.02214076e+23
```

Comment:
Take constants from NIST and document each. Eg
```python
#: Boltzmann's constant :math:`k_B` in kJ/(mol K); value from `NIST CODATA: k<https://physics.nist.gov/cgi-bin/cuu/Value?k>`_.
k_Boltzmann = 1.380649e-23
```
Also use expressive variable names.